trie/statistics/trieStatisticsCollector.go

```diff
@@ -60,18 +45,13 @@ func (tsc *trieStatisticsCollector) Add(trieStats common.TrieStatisticsHandler)
 	tsc.mutex.Lock()
 	defer tsc.mutex.Unlock()
 
-	tsc.numNodes += trieStats.GetTotalNumNodes()
-	tsc.triesSize += trieStats.GetTotalNodesSize()
-	tsc.numDataTries++
-
-	tsc.numTotalBranches += trieStats.GetNumBranchNodes()
-	tsc.numTotalExtensions += trieStats.GetNumExtensionNodes()
-	tsc.numTotalLeaves += trieStats.GetNumLeafNodes()
-	tsc.totalSizeBranches += trieStats.GetBranchNodesSize()
-	tsc.totalSizeExtensions += trieStats.GetExtensionNodesSize()
-	tsc.totalSizeLeaves += trieStats.GetLeafNodesSize()
-	for version, numNodes := range trieStats.GetLeavesMigrationStats() {
-		tsc.migrationStats[version] += numNodes
+	if _, ok := tsc.trieStatsByType[trieType]; !ok {
+		tsc.trieStatsByType[trieType] = NewTrieStatistics()
+	}
+
+	tsc.trieStatsByType[trieType].MergeTriesStatistics(trieStats)
+	if trieType == common.DataTrie {
```

Comment:
this is a bit ugly. Could have used a map of type common.TrieType and print everything on the log line on L87. In this way the trieStatisticsCollector will not care about the the fact that we have a special type of common.TrieType. I've done something similar in p2p package that will output a log like:
... observe the `observers histogram` part...
```
DEBUG[2023-05-11 11:33:48.163]   network connection status                known peers = 46 connected peers = 28 intra shard validators = 7 intra shard observers = 1 cross shard validators = 15 cross shard observers = 4 full history observers = 0 unknown = 0 seeders = 1 current shard = 0 validators histogram = shard 0: 7, shard 1: 4, shard 2: 6, meta: 5 observers histogram = shard 0: 1, shard 1: 1, shard 2: 1, meta: 2 preferred peers histogram =  
```

https://github.com/multiversx/mx-chain-p2p-go/blob/b6fe55be463cdd67bf119d9e1f7eacc2da3053ad/libp2p/netMessenger.go#L575

(the whole print in the netMessenger can be refactored at some point to abide by this simplification)

Reply:
Refactored.